vm/array_test.go

```diff
@@ -205,6 +205,77 @@ func TestArrayPlusOperatorFail(t *testing.T) {
 	}
 }
 
+func TestArrayAnyMethod(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected interface{}
+	}{
+		{`
+			[1, 2, 3].any? do |e|
+			  e == 2
+			end
+		`, true},
+		{`
+			[1, 2, 3].any? do |e|
+			  e
+			end
+		`, true},
+		{`
+			[1, 2, 3].any? do |e|
+			  e == 5
+			end
+		`, false},
+		{`
+			[1, 2, 3].any? do |e|
+			  nil
+			end
+		`, false},
+	}
+
+	for i, tt := range tests {
+		v := initTestVM()
+		evaluated := v.testEval(t, tt.input, getFilename())
+		checkExpected(t, i, evaluated, tt.expected)
+		v.checkCFP(t, i, 0)
+		v.checkSP(t, i, 1)
+	}
+}
+
+func TestArrayAnyMethodBlockNotCalled(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected interface{}
+	}{
+		{`
+			[].any? do |e|
+			  true
+			end
+		`, false},
+	}
+
+	for i, tt := range tests {
+		v := initTestVM()
+		evaluated := v.testEval(t, tt.input, getFilename())
+		checkExpected(t, i, evaluated, tt.expected)
+		v.checkCFP(t, i, 1)
```

Comment:
@st0012 I'm not sure about this. Originally I thought this was an issue (I don't how exactly what CFP is), then I found that `Array#select` yield the same result (in testing) if a test case with an empty block is performed.

Reply:
What do you mean yield the same result? You mean also executes the block?

Reply:
And I know why the CFP is `1` instead of `0`.
The CFP stands for call frame pointer. In Goby we separate value stack and method call stack. And when we having a block argument, a block's call frame will be pushed on the call stack. And we pop that block's call frame after we execute (`yield`) the block.
But in this case you give it an empty array, which means the for loop won't be evaluated. So no yield action will be performed, the block's call frame just stays there.

Reply:
BTW, thanks for spotting this issue, you can see how I fix this issue in [here](https://github.com/goby-lang/goby/commit/e3372b7306273e6c6e8937d69d8ee42e75932631)

Reply:
Thanks! Fixed, and updated the UT.